packages/coordinator/src/services/verification/certificate-revocation/verify.ts

```diff
@@ -0,0 +1,138 @@
+ * @returns true if the certificate is considered revoked, false otherwise
+ */
+const isCertificateRevoked = (
+  cert: RevokedCertificate,
+  prescriptionSignedDate: Date,
+  logger: pino.Logger
+): boolean => {
+  const certSerialNumber = getRevokedCertSerialNumber(cert)
+  const signedAfterRevocation = wasPrescriptionSignedAfterRevocation(prescriptionSignedDate, cert)
+  const errorMsgPrefix = `Certificate with serial '${certSerialNumber}' found on CRL`
+
+  const reasonCode = getRevokedCertReasonCode(cert)
+  if (!reasonCode) {
+    logger.error(`Cannot extract Reason Code from CRL for certificate with serial ${certSerialNumber}`)
+    return signedAfterRevocation // prescription invalid if signed after revocation
+  }
+
+  switch (reasonCode) {
+    // AEA-2650 - AC 1.1
+    case CRLReasonCode.Unspecified:
+    case CRLReasonCode.AffiliationChanged:
+    case CRLReasonCode.Superseded:
+    case CRLReasonCode.CessationOfOperation:
+    case CRLReasonCode.CertificateHold:
+    case CRLReasonCode.RemoveFromCRL:
+      if (signedAfterRevocation) logger.warn(`${errorMsgPrefix} with Reason Code ${reasonCode}`)
+      return signedAfterRevocation
+
+    // AEA-2650 - AC 1.2
+    case CRLReasonCode.KeyCompromise:
+    case CRLReasonCode.CACompromise:
+      logger.warn(`${errorMsgPrefix} with Reason Code ${reasonCode}`)
+      return true // always consider prescription invalid
+
+    default:
+      if (signedAfterRevocation) logger.warn(`${errorMsgPrefix} with unhandled Reason Code ${reasonCode}`)
+      return signedAfterRevocation
+  }
+}
+
+type CertData = {
+  certificate: X509,
+  serialNumber: string
+}
+
+const parseCertificateFromPrescription = (parentPrescription: hl7V3.ParentPrescription): CertData => {
+  const certificate = getCertificateFromPrescription(parentPrescription)
+  const serialNumber = getX509SerialNumber(certificate)
+  return {certificate, serialNumber}
+}
+
+const isSignatureCertificateValid = async (
+  parentPrescription: hl7V3.ParentPrescription,
+  logger: pino.Logger
+): Promise<boolean> => {
+  const prescriptionSignedDate = getPrescriptionSignatureDate(parentPrescription)
+  const prescriptionId = getPrescriptionId(parentPrescription)
+  const {certificate, serialNumber} = parseCertificateFromPrescription(parentPrescription)
+
+  if (!certificate) {
+    logger.error(`Could not parse X509 certificate from prescription with ID '${prescriptionId}'`)
+    return false
+  }
+
+  const distributionPointsURI = certificate?.getExtCRLDistributionPointsURI()
+  if (!distributionPointsURI || distributionPointsURI.length === 0) {
+    logger.error(`Cannot retrieve CRL distribution point from certificate with serial ${serialNumber}`)
+    return true // TODO: Add decision log number with justification
+  }
+
+  // Loop through the Distribution Points found on the cert
+  for (const distributionPointURI of distributionPointsURI) {
+    const crl = await getRevocationList(distributionPointURI)
```

Comment:
Is there a way to validate CRL has been issued by the CA?

Reply:
If necessary, CRL validation will be done as part of https://nhsd-jira.digital.nhs.uk/browse/AEA-2998